### DISCUSSION TITLE
Getting transaction reference in Sea-orm assigned to struct variable

### DISCUSSION THREAD

**Discussion by 0xvasanth** [SeaQL/sea-orm#971] [Q&A]
Getting transaction reference in Sea-orm assigned to struct variable to handle the commit or rollback in future based on the consumer decision 

Example:

this is a [custom session](https://github.com/itsparser/orca/blob/lifetime/base/src/client/database/session.rs) created in our repo
it has lifetime of its own connection will be created outside and moved in this struct `new` function 

now when I start the `commit` function 
now unable to run the commit 

<img width="960" alt="Screenshot 2022-08-15 at 2 38 57 PM" src="https://user-images.githubusercontent.com/13918750/184608392-65e24c0d-bc9a-4bf6-879f-24f7574adc1f.png">

**Comment by 0xvasanth** [comment]
<img width="960" alt="Screenshot 2022-08-15 at 2 41 46 PM" src="https://user-images.githubusercontent.com/13918750/184608680-6c1422e8-f0ac-4e66-8c0f-95b9bb78dc01.png">

[code link](https://github.com/itsparser/orca/blob/lifetime/base/src/client/database/session.rs)

**Comment by billy1624** [comment]
Hey @itsparser, if you put a transaction behind an `Option<T>` when you commit the transaction. You should take a mutable borrow of it instead of unwrapping it.

Btw... I think it would be much simple to begin a transaction directly inside the controller then borrow it down to the handler for CRUD operations. Finally, it's the owner of the transaction (at controller level) to commit / rollback it.
